File: app/src/main/java/com/riders/thelab/core/broadcast/ConnectivityReceiver.java
package com.riders.thelab.core.broadcast;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.riders.thelab.core.utils.LabNetworkManager;


public class ConnectivityReceiver extends BroadcastReceiver {

    private static ConnectivityReceiver mInstance;

    private final ConnectivityReceiverListener connectivityReceiverListener;

    public ConnectivityReceiver() {
        connectivityReceiverListener = null;
    }

    public ConnectivityReceiver(ConnectivityReceiverListener connectivityReceiverListener) {
        super();
        this.connectivityReceiverListener = connectivityReceiverListener;
    }

    public static ConnectivityReceiver getInstance() {
        if (null == mInstance) {
            mInstance = new ConnectivityReceiver();
        }

        return mInstance;
    }

    @SuppressLint("NewApi")
    @Override
    public void onReceive(Context context, Intent arg1) {

        if (LabCompatibilityManager.isLollipop()) {

            boolean isConnected = LabNetworkManager.isConnected(context);

            if (connectivityReceiverListener != null) {
                connectivityReceiverListener.onNetworkConnectionChanged(isConnected);
            }
        }
    }

    public ConnectivityReceiverListener getConnectivityReceiverListener() {
        return connectivityReceiverListener;
    }

    public interface ConnectivityReceiverListener {
        void onNetworkConnectionChanged(boolean isConnected);
    }
}

File: app/src/main/java/com/riders/thelab/core/broadcast/ConnectivityReceiver.kt
package com.riders.thelab.core.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.riders.thelab.core.utils.LabNetworkManagerNewAPI

class ConnectivityReceiver(
    private val mConnectivityReceiverListener: ConnectivityReceiverListener?
) : BroadcastReceiver() {

    companion object {
        private var mInstance: ConnectivityReceiver? = null

        fun getInstance(): ConnectivityReceiver {
            if (null == mInstance) {
                this.mInstance = ConnectivityReceiver()
            }
            return this.mInstance as ConnectivityReceiver
        }
    }

    constructor() : this(null)

    override fun onReceive(context: Context?, intent: Intent?) {
        var isConnected = false
        if (null != context) {
            isConnected = LabNetworkManagerNewAPI.getInstance(context).isOnline()
        }
        mConnectivityReceiverListener?.onNetworkConnectionChanged(isConnected)
    }

    fun getConnectivityReceiverListener(): ConnectivityReceiverListener? {
        return mConnectivityReceiverListener
    }

    interface ConnectivityReceiverListener {
        fun onNetworkConnectionChanged(isConnected: Boolean)
    }
}